content/post/introducing-custom-container/index.md

```diff
@@ -0,0 +1,165 @@
+tags:
+  - kubeDB
+  - container
+  - MySQL
+  - Filebeat
+  - StatefulSet
+  - MySQLOpsRequest
+---
+
+### Add custom containers in KubeDB managed Database
+
+Let's assume you have a KubeDB managed Database deployed in your Kubernetes environment. Now, You want to inject a sidecar container in the database StatefulSet in order to extend and enhance the functionality of existing containers. Currently, KubeDB doesn't have support for custom container insertion. But, we are having a workaround to run KubeDB managed pods with your own custom container. In this demo, we are going to demonstrate how to inject Filebeat container in KubeDB managed MySQL cluster.  
+
+Let's assume you have a KubeDB managed MySQL cluster deployed in your Kubernetes environment,  Here's a quick demonstration on how to accomplish it.
+
+Firstly, We are going to deploy a sample MySQL cluster in demo namespace using the following YAML.
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: MySQL
+metadata:
+  name: mysql-cluster
+  namespace: demo
+spec:
+  replicas: 2
+  version: "8.0.29"
+  storageType: Durable
+  storage:
+    storageClassName: "standard"
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+  terminationPolicy: WipeOut
+```
+
+The Operator will create a custom resource `mysql-cluster`. It will also create a StatefulSet named `mysql-cluster` as well. The StatefulSet will eventually create two pods named `mysql-cluster-0` and `mysql-cluster-1`.
+Our aim is to use [Filebeat](https://www.elastic.co/beats/filebeat) to retrieve database logs. We are going to inject a filebeat container with custom configuration to the operator-generated mysql-cluster [StatefulSets](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset).
+### Prepare the container
+
+At first, create a filebeat.yml file with your required configurations. Let's assume that you want to read DB logs from a mysql container. In kubeDB managed MySQL pod, the log file is mounted in `/var/lib/mysql` directory in `{pod-name}.log` file.
+
+The following example configures Filebeat to harvest lines from all log files that match the specified glob patterns. For this demo, we are going to Forward the output to console. The Console output writes events in JSON format to stdout.
+
+```yaml
+filebeat.inputs:
+  - type: log
+    paths:
+    - /var/lib/mysql/*.log
+output.console:
+  pretty: true
+```
+
+You can also provide specific log file paths for specific pods as well. Use the following command to view all the log files available in `mysql-cluster-0` pod.
+
+```bash
+$ kubectl exec -it mysql-cluster-0 -n demo -c mysql -- bash -c "cd /var/lib/mysql && ls | grep *.log"
+
+mysql-quickstart-0.log
+```
+
+Now create a Dockerfile in the same directory. Set the base image to your required filebeat version. For this demo, we are using `Filebeat 7.17.1`. In the Dockerfile, copy the filebeat.yml to default home directory to override the existing one.
```

Comment:
Since we cannot mount additional volumes, let's move the configuration to the docker image instead.